overlord/snapstate/backend/snapdata.go

```diff
@@ -112,14 +111,9 @@ func copySnapData(oldSnap, newSnap *snap.Info) (err error) {
 func copySnapDataDirectory(oldPath, newPath string) (err error) {
 	if _, err := os.Stat(oldPath); err == nil {
 		if _, err := os.Stat(newPath); err != nil {
-			// there is no golang "CopyFile"
-			cmd := exec.Command("cp", "-a", oldPath, newPath)
-			if output, err := cmd.CombinedOutput(); err != nil {
-				output = bytes.TrimSpace(output)
-				if len(output) > 0 {
-					err = fmt.Errorf("%s", output)
-				}
-				return fmt.Errorf("cannot copy %s to %s: %v", oldPath, newPath, err)
+			err := osutil.CopyFile(oldPath, newPath, osutil.CopyFlagPreserveAll|osutil.CopyFlagSync)
```

Comment:
This and the following can be a single line: `if err := osutil... ; err != nil {`